ngsderive/commands/instrument.py

```diff
@@ -227,22 +227,34 @@ def main(ngsfiles, outfile=sys.stdout, n_samples=10000):
         malformed_read_names = False
 
         # accumulate instrument and flowcell IDs
-        for read in itertools.islice(ngsfile, n_samples):
-            parts = read["query_name"].split(":")
-            if len(parts) != 7:  # not Illumina format
-                malformed_read_names = True
-                iid = parts[0]  # attempt to recover machine name
+        try:
+            for read in itertools.islice(ngsfile, n_samples):
+                parts = read["query_name"].split(":")
+                if len(parts) != 7:  # not Illumina format
+                    malformed_read_names = True
+                    iid = parts[0]  # attempt to recover machine name
+                    instruments.add(iid)
+                    for rg in ngsfile.handle.header.to_dict()["RG"]:
```

Comment:
Should this wrap this line, since this is the line we're trying to access the key?

Reply:
That's what I thought at first as well, but
```
for read in itertools.islice(ngsfile, n_samples):
```
is where the `KeyError` first gets thrown. That's because during the underlying `NGSFile.__next__()` call the `islice()` causes, the RG tag of the individual reads are accessed.